client/state/happychat/utils.js

```diff
@@ -0,0 +1,62 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import config from 'config';
+import { getGroups } from 'state/happychat/selectors';
+import { getCurrentUser, getCurrentUserLocale } from 'state/current-user/selectors';
+import { getHelpSelectedSite } from 'state/help/selectors';
+
+// Promise based interface for wpcom.request
+const request = ( ...args ) =>
+	new Promise( ( resolve, reject ) => {
+		wpcom.request( ...args, ( error, response ) => {
+			if ( error ) {
+				return reject( error );
+			}
+			resolve( response );
+		} );
+	} );
+
+const sign = payload =>
+	request( {
+		method: 'POST',
+		path: '/jwt/sign',
+		body: { payload: JSON.stringify( payload ) },
+	} );
+
+const startSession = () =>
+	request( {
+		method: 'POST',
+		path: '/happychat/session',
+	} );
+
+export default state => () => {
```

Comment:
Giving the fact that this is more of a generic file now, could we export named methods instead of having getAuth as the default for utils.

Reply:
Fixed.